docs/DeveloperGuide.md

```diff
@@ -69,20 +69,120 @@ The sections below give more details of each component.
 
 ![Structure of the UI Component](images/UiClassDiagram.png)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `StudentListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
 
-The `UI` component uses the JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/AY2223S2-CS2103T-T15-4/tp/blob/master/src/main/java/tfifteenfour/clipboard/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/AY2223S2-CS2103T-T15-4/tp/blob/master/src/main/resources/view/MainWindow.fxml)
+The UI also consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `HelpWindow`, and `StatusBarFooter`. All these, including the `MainWindow`, inherit from the abstract `UiPart` class which captures the commonalities between classes that represent parts of the visible GUI.
+These parts also remain visible as the user navigates through the different pages in the GUI.
+
+#### Initialising the UI
+
+![Homepage](images/UiHomePage.png)
+
+*Figure 2: CLIpboard landing page*
+
+The `UiManager` is a controller class containing the `start` method which initialises the UI in the following sequence:
+
+1. Instantiate a new `MainWindow` object.
+2. `show` the `MainWindow` object.
+3. Populate the UI window using the the `fillInnerParts` method of `MainWindow`.
+
+The `MainWindow#fillInnerParts` method will populate the UI by:
+1. Populating the Left Pane with a `CourseListPanel`.
+2. Populating the `pageTab`
+3. Populating the `commandBox`
+4. Populating the `resultDisplay`
+5. Populating the `statusBarFooter`
+
+As shown in Figure 2, users will be shown the Course Page upon launching the application. The GUI will then refresh accordingly as the user navigates through the different pages in the application.
+
+#### Navigating through pages
+
+![Navigation Guide](images/navigation.png)
+
+*Figure 3: CLIpboard navigation guide*
+
+As shown in Figure 3, users can navigate through the application using a combination of `select`, `task`, `session`, and `back` commands.
+When a command such as `select` is called, the left pane or the right pane, or both, will be refreshed in `MainWindow`.
+
+
+![Select Sequence Diagram](images/SelectSequenceDiagram.png)
+
+*Figure 4: Sequence Diagram for `select`*
+
+
+Figure 4's Sequence Diagram depicts how the `UI`, `Logic`, and `Model` components interact when `select 1` is called on the Course Page. The user's selection path (`Course` -> `Group` -> `Student`) is tracked by the `CurrentSelection` class as they navigate through the application.
```

Comment:
> The user's selection path (`Course` -> `Group` -> `Student`)

is this thing mentioned anywhere else? I think might be confusing at first to understand what does this mean as the example is only showing (Course -> Group)🤔

Reply:
Hmm for this case would (`Course` -> `Group`) be clearer?